### ISSUE TITLE
[OpenCV] unable to build for x64-windows using v141 toolset

### ISSUE THREAD

**Issue by cenit** [Microsoft/vcpkg#1855]
Again, due to #1619, OpenCV is now broken also on x64-windows with my toolchain (v141), because a depencency (ceres) can only be built using VS 2015 (which is not available on my machine), due to this line in the portfile (line 19 in ports/ceres/portfile.cmake)
```
set(VCPKG_PLATFORM_TOOLSET "v140") # Force VS2015 because VS2017 compiler return internal error
```
To be honest, OpenCV 3.3 PR has been stagnant for so long I was hoping because of tests, but now that it is merged OpenCV is impossible to build, in any configuration...
Did you tested it? Or breaking previous available triplets without any notice is an intended effect?
Isn't this port mechanism (instead of an automatic system) in place just to avoid these problems?

**Comment by jasjuang** [comment]
I only tested in VS 2015 instead of VS 2017. The ceres dependency is from opencv_contrib. You can temporarily go into the portfile to disable opencv_contrib and it should build.

**Comment by ras0219-msft** [comment]
Thank you very much for opening this issue. I've disabled the ceres dependency for all architectures for now; we'll look at conditionally re-enabling it for x64 only once we're able to fix the VS2017 incompatibility.

**Comment by ras0219-msft** [comment]
There are still some lingering issues which we are actively working to resolve.

**Comment by ras0219-msft** [comment]
Ok, all the above builds should be fixed in the latest master (3.3.0-4).

We will test OpenCV more in the future to avoid issues like this, especially considering how widely used it is.

**Comment by lazylazypig** [comment]
@ras0219-msft hi, before install opencv, I have installed eigen3 (the newest version 3.3.4-1) from the port. when I install opencv, the opencv_sfm module failed with the flowing errors:

       “D:\Development\Env\vcpkg\buildtrees\opencv\x64-windows-rel\install.vcxproj”(default target) (1) ->
       “D:\Development\Env\vcpkg\buildtrees\opencv\x64-windows-rel\ALL_BUILD.vcxproj”(default target) (3) ->
       “D:\Development\Env\vcpkg\buildtrees\opencv\x64-windows-rel\modules\sfm\src\libmv_light\libmv\multiview\multiview.vcxproj”(default target) (12) ->
       (ClCompile target) -> 
         d:\development\env\vcpkg\installed\x64-windows\include\eigen\src\core\redux.h(237): fatal error C1001: An internal error has occurred in the compiler. [D:\Development\Env\vcpkg\buildtrees\opencv\x64-windows-rel\modules\sfm\src\libmv_light\libmv\multiview\multiview.vcxproj]

    347 Warning(s)
    1 Error(s)

as a result, I have to disable the opencv_sfm module by adding "-DBUILD_opencv_sfm=OFF" option  in my portfile.cmake or removing eigen3.3.4 from my vcpkg environment to make opencv build successfully.

but, I'll be very interested in how to fix this issue? thanks:)
